### ISSUE TITLE
Train the Model

### ISSUE THREAD

**Issue by pffaundez** [Akshay-Rajesh/Spam-text-classifier#8]

**Comment by Akshay-Rajesh** [comment]
First training completed , below are the results 👍 
accuracy
0.9919282511210762
f1_score
0.9688581314878892
precision
1
recall
0.9395973154362416

**Comment by Akshay-Rajesh** [comment]
Model looks overfitted , needs more clenaing .

**Comment by Akshay-Rajesh** [comment]
Trained the Naive bayes model using the preprocessed data . Below are the results .

**Comment by Akshay-Rajesh** [comment]
![Image](https://github.com/user-attachments/assets/e94698bc-cb0a-40f6-b103-39fb072ff249)

**Comment by Akshay-Rajesh** [comment]
results updated in Mlflow

**Comment by Akshay-Rajesh** [comment]
Trained an SVM model :
Accuracy: 0.8905829596412556
Precision: 0.5652173913043478
Recall: 0.785234899328859
F1 Score: 0.6573033707865169

**Comment by pffaundez** [comment]
Great @Akshay-Rajesh! 
We, I think, are in a decent performance of the NB multinomial model for binary classification. Did we check overfitting of the model through CV or so?

**Comment by Akshay-Rajesh** [comment]
Accuracy shows a deviation in cross validation :

![Image](https://github.com/user-attachments/assets/dc757889-65b6-44cb-ba34-88bf646b9442)

**Comment by Akshay-Rajesh** [comment]
I have added a small hyperparameter tuning script in main.py , this improved the precision and recall . Results below 

![Image](https://github.com/user-attachments/assets/703386bb-3270-4ca7-8a3a-0fc3daa5ab75)

**Comment by Akshay-Rajesh** [comment]
results after hyperparameter tuning
test_accuracy
0.979372197309417
test_f1_score
0.9235880398671097
test_precision
0.9144736842105263
test_recall
0.9328859060402684
results before tuning hyperparameters
test_accuracy
0.9757847533632287
test_f1_score
0.9120521172638436
test_precision
0.8860759493670886
test_recall
0.9395973154362416

**Comment by Akshay-Rajesh** [comment]
After splitting the data as 70-30 , results are as below .

![Image](https://github.com/user-attachments/assets/be4ab6ba-76db-4620-a748-d51845b43fc2)

**Comment by Akshay-Rajesh** [comment]
Results with confusion matrix ,

Test Set Evaluation:
Accuracy: 0.9742822966507177
Precision: 0.9751196031561967
Recall: 0.9742822966507177
F1 Score: 0.9745869276605553
Confusion Matrix:
[[1420   28]
 [  15  209]]
